sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/models/SnapshotSelector.java

```diff
@@ -55,4 +61,56 @@ public SnapshotSelector setSnapshotStatus(SnapshotStatus... status) {
         return this;
     }
 
+    /**
+     * Gets the fields on {@link ConfigurationSettingsSnapshot} to return from the GET request. If none are set, the
+     * service returns the snapshot with all of their fields populated.
+     *
+     * @return The set of {@link ConfigurationSettingsSnapshot} fields to return for a GET request.
+     */
+    public List<SnapshotFields> getFields() {
+        return fields;
+    }
+
+    /**
+     * Sets fields that will be returned in the response corresponding to properties in
+     * {@link ConfigurationSettingsSnapshot}. If none are set, the service returns snapshot with all of their fields
+     * populated.
+     *
+     * @param fields The fields to select for the query response. If none are set, the service will return the
+     * snapshot with a default set of properties.
+     *
+     * @return The updated SnapshotSelector object.
+     */
+    public SnapshotSelector setFields(SnapshotFields... fields) {
+        this.fields = fields == null ? null : Arrays.asList(fields);
+        return this;
+    }
+
+    @Override
+    public String toString() {
+        String fields;
+        if (CoreUtils.isNullOrEmpty(this.fields)) {
+            fields = "ALL_FIELDS";
+        } else {
+            // join a list of enum values into a comma-separated string
+            fields = this.fields == null ? null : IterableStream.of(this.fields)
+                .stream()
+                .map(fieldsEnumValue -> fieldsEnumValue.toString())
+                .collect(Collectors.joining(","));
+        }
+
+        String status;
+        if (CoreUtils.isNullOrEmpty(this.status)) {
+            status = "ALL_STATUS";
+        } else {
+            // join a list of enum values into a comma-separated string
+            status = this.status == null ? null : IterableStream.of(this.status)
+                .stream()
+                .map(statusEnumValue -> statusEnumValue.toString())
+                .collect(Collectors.joining(","));
+        }
+
+        return String.format("SnapshotSelector(name=%s, status=%s, fields=%s)",
```

Comment:
Remove the String.format and just use String concatenation. Another option would be using a StringBuilder for the entirety of this.